### PR TITLE
Add Mergify integration to automatically merge PRs on approval

### PR DESCRIPTION
**Pull request by PrasadG193**

### CODE REVIEW

**Comment by sahil-lakhwani** [comment]
Does this not wait for the next reviewers after one has approved? I am just thinking what happens when you want more than one reviewers' approval.

**Comment by PrasadG193** [comment]
@sahil-lakhwani you need to add `hold-off-merging` label to hold off merging

**Comment by PrasadG193** [comment]
@sahil-lakhwani we can also add a condition if a particular label is added only then merge the PR. Let me do that

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+pull_request_rules:
+  - name: Automatic merge on approval
+    conditions:
+      - base=main
+      - "#approved-reviews-by>=1"
+      - label!=hold-off-merging
```
Comment (PrasadG193) [inline .mergify.yml:6]:
```suggestion
      - label!=hold-off-merging
      - label=ready-to-merge
```

**Pull request merged**